source/s3_auto_ranged_get.c

```diff
@@ -266,6 +282,8 @@ static bool s_s3_auto_ranged_get_update(
                     request->part_number,
                     &request->part_range_start,
                     &request->part_range_end);
+                
+                request->ticket = ticket;
```

Comment:
trivial: it's a little spooky that this assignment happens 15 lines after the `ticket` is created. the `ticket` would leak if someone added an early-out branch in the interim.

why not just create the ticket directly into the member variable? like:
```
request->ticket = aws_s3_buffer_pool_reserve(...);
if (request->ticket == NULL)
```

Reply:
moved it a bit up. 
allocating into request is a bit of a chicken and egg problem. if reservation fails we dont want to create a request at all